packages/namekit-react/src/components/PriceInfo.tsx

```diff
@@ -0,0 +1,80 @@
+import React, { SVGProps } from "react";
+import { Tooltip } from "@namehash/nameguard-react";
+import {
+  getPriceDescription,
+  Registration,
+  DomainName,
+} from "@namehash/ens-utils";
+
+interface PriceInfoProps {
+  parsedName: DomainName;
+  registration: Registration;
+  // Wether to display the icon with a tooltip or the icon + the price info text directly
+  displayIconWithTooltip?: boolean;
+}
+
+export const PriceInfo = ({
+  displayIconWithTooltip = true,
+  registration,
+  parsedName,
+}: PriceInfoProps) => {
+  const priceDescription = getPriceDescription(registration, parsedName);
+
+  if (!priceDescription) return <></>;
+
+  const priceDescriptionElm = (
+    <div className="font-normal inline">
+      <p className="inline">{priceDescription.descriptiveTextBeginning}</p>
+      <p className="font-medium inline">
+        {priceDescription.pricePerYearDescription}
+      </p>
+      <p className="inline">{priceDescription.descriptiveTextEnd}</p>
+    </div>
+  );
+
+  /*
+    If only an icon must be shown in the UI, we return the icon and when the user hovers over it, the info tooltip will be shown.
+    Otherwise, we return the icon and the text in the UI directly.
+  */
+  if (displayIconWithTooltip)
+    return (
+      <Tooltip
+        trigger={
+          <InfoFilledIcon className="h-5 w-5 text-gray-300 transition hover:text-gray-400 cursor-pointer" />
+        }
+      >
+        <p className="text-white">{priceDescriptionElm}</p>
+      </Tooltip>
+    );
+  else
+    return (
+      <div className="flex items-center flex-row mb-4">
+        <div className="h-5 w-5">
+          <InfoFilledIcon className="h-5 w-5 text-gray-300" />
+        </div>
+        <p className="text-sm text-gray-500 ml-3">
+          <div className="inline mr-2">{priceDescriptionElm}</div>
+        </p>
+      </div>
+    );
+};
+
+export function InfoFilledIcon(props: SVGProps<SVGSVGElement>) {
```

Comment:
Can we move this into a separate .tsx file that's dedicated just for this icon?

Also, is there any special reason we call this "InfoFilledIcon"? Why is it "Filled"? Is there any special reason not to just call it "InfoIcon"?

Reply:
Renamed and moved this component 🙂👍🏼